### DISCUSSION TITLE
Best practice to handle a scenario where users belong to distinct groups

### DISCUSSION THREAD

**Discussion by superseby2** [appwrite/appwrite#2783] [Q&A]
Our webapp targets 2 groups of users.  A & B. 

We want to use LinkedIN Oauth only to allow users to signin and signup and want to have 2 really separated group of users. 

The problem I have is to  be able to affect one user to the proper group at signup. 

So far the solution I could came with is 

- users A login using /app/a and users B /app/b
- when coming back from oauth we know which group they belong to because of the redirect path they came from 
- we update the user preferences to add a mention of his/her group using the user session
- we will use user's preferences to make sure a logged-in user belonging to A cannot access a the authenticated area from group B

The solution is far from being optimal  .. and would love to hear if you see some other path I should follow. 

Thx !

**Comment by superseby2** [comment]
A nice option would be to be able to call /v1/account/sessions/oauth2/{provider} with user preferences attached that would be used once the user is created. But I could not see such an option

**Comment by eldadfux** [comment]
Are you using 2 separated Appwrite projects? Why not use a single one and just add the group info in the user prefs?
